runtime/legion/legion_views.cc

```diff
@@ -1550,7 +1618,65 @@ namespace Legion {
       }
       if (parent != NULL)
         parent->collect_users(term_events);
-    } 
+    }
+
+    //--------------------------------------------------------------------------
+    void MaterializedView::collect_traced_users()
```

Comment:
I'm not really happy with this kind of architecture. I think views should be pretty ignorant of tracing and shouldn't really be responsible for remembering things about tracing. That's not really their job and I want them to be as unencumbered as possible since their multi-node implementations are already complex as it is.